recipe/build.sh

```diff
@@ -106,9 +97,89 @@ export TF_DOWNLOAD_CLANG=0
 export TF_SET_ANDROID_WORKSPACE=0
 export TF_CONFIGURE_IOS=0
 
-# Get rid of unwanted defaults
-sed -i -e "/PROTOBUF_INCLUDE_PATH/c\ " .bazelrc
-sed -i -e "/PREFIX/c\ " .bazelrc
+## cuda settings
+if [[ ${cuda_compiler_version} != "None" ]]; then
+
+    sed -i -e "s:\${PREFIX}:${PREFIX}:" tensorflow/core/platform/default/build_config/BUILD
+
+    export CUDA_TOOLKIT_PATH=/usr/local/cuda-${cuda_compiler_version}
+    export TF_CUDA_PATHS="${PREFIX},/usr/local/cuda-${cuda_compiler_version},/usr"
+    export USE_CUDA=1
+    export cuda=Y
+    if [[ ${cuda_compiler_version} == 9.0* ]]; then
+        export TF_CUDA_COMPUTE_CAPABILITIES=5.2,5.3,6.0,6.1,6.2
+    elif [[ ${cuda_compiler_version} == 9.2* ]]; then
+        export TF_CUDA_COMPUTE_CAPABILITIES=5.2,5.3,6.0,6.1,6.2,7.0
+    elif [[ ${cuda_compiler_version} == 10.* ]]; then
+        export TF_CUDA_COMPUTE_CAPABILITIES=5.2,5.3,6.0,6.1,6.2,7.0,7.2,7.5
+    elif [[ ${cuda_compiler_version} == 11.0* ]]; then
+        export TF_CUDA_COMPUTE_CAPABILITIES=5.2,5.3,6.0,6.1,6.2,7.0,7.2,7.5,8.0
+    elif [[ ${cuda_compiler_version} == 11.1 ]]; then
+        export TF_CUDA_COMPUTE_CAPABILITIES=5.2,5.3,6.0,6.1,6.2,7.0,7.2,7.5,8.0,8.6
+    elif [[ ${cuda_compiler_version} == 11.2 ]]; then
+        export TF_CUDA_COMPUTE_CAPABILITIES=5.2,5.3,6.0,6.1,6.2,7.0,7.2,7.5,8.0,8.6
+    else
+        echo "unsupported cuda version. edit build_pytorch.sh"
+        exit 1
+    fi
+    
+    export TF_NEED_CUDA=1
+    export TF_CUDA_VERSION="${cuda_compiler_version}"
+    export TF_CUDNN_VERSION="${cudnn}"
+    export TF_CUDA_CLANG=0
+    export TF_DOWNLOAD_CLANG=0
+    export TF_NEED_TENSORRT=0
+    export GCC_HOST_COMPILER_PATH="${CC}"
+    export TF_NCCL_VERSION=""
+    export CC_OPT_FLAGS="-march=nocona -mtune=haswell"
+    export GCC_HOST_COMPILER_PATH="${CC}"
+    
+    BUILD_OPTS="
+    --copt=-march=nocona
+    --copt=-mtune=haswell
+    --copt=-ftree-vectorize
+    --copt=-fPIC
+    --copt=-fstack-protector-strong
+    --copt=-O2
+    --cxxopt=-fvisibility-inlines-hidden
+    --cxxopt=-fmessage-length=0
+    --linkopt=-zrelro
+    --linkopt=-znow
+    --linkopt=-L${PREFIX}/lib
+    --verbose_failures
+    --config=opt
+    --config=cuda
+    --strip=always
+    --color=yes
+    --curses=no
+    --action_env=PYTHON_BIN_PATH=${PYTHON}
+    --action_env=PYTHON_LIB_PATH=${SP_DIR}
+    --python_path=${PYTHON}
+    --define=PREFIX=$PREFIX
+    --copt=-DNO_CONSTEXPR_FOR_YOU=1
+    --host_copt=-DNO_CONSTEXPR_FOR_YOU=1
+    --define=LIBDIR=$PREFIX/lib
+    --define=INCLUDEDIR=$PREFIX/include"
+
+    ## Doesn't work with CUDA builds, don't know why
+    export TF_SYSTEM_LIBS=""
```

Comment:
What's the error you're getting?

Reply:
It wasn't finding headers, e.g. I would get

```
Execution platform: @local_execution_config_platform//:platform
In file included from tensorflow/core/platform/protobuf.cc:16:
./tensorflow/core/platform/protobuf.h:28:10: fatal error: google/protobuf/io/coded_stream.h: No such file or directory
   28 | #include "google/protobuf/io/coded_stream.h"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
```

I believe this was a consequence of not using `custom_tookchain`. Now that I'm back on track with that I no longer get these errors.